projects/core/components/dialog/dialog.service.ts

```diff
@@ -11,7 +11,9 @@ const DIALOG = new PolymorpheusComponent(TuiDialogComponent);
 @Injectable({
     providedIn: `root`,
 })
-export class TuiDialogService extends AbstractTuiDialogService<TuiDialogOptions<any>> {
+export class TuiDialogService<T = any> extends AbstractTuiDialogService<
```

Comment:
Let's not change that. If you want to type options – you can always create a typed const before passing it to the `open` method. Typing the entire service is just wrong, because the service itself supports all the options, it's a matter of the content you passed. I tried to improve typing there but failed so far 😢 
https://github.com/taiga-family/taiga-ui/pull/4170

Reply:
fixed